sqlglot/parser.py

```diff
@@ -3829,8 +3826,42 @@ def _parse_merge(self) -> exp.Expression:
             expressions=whens,
         )
 
-    def _parse_set(self) -> exp.Expression:
-        return self.expression(exp.Set, expressions=self._parse_csv(self._parse_set_item))
+    def _parse_set(self):
+        start = self._prev
+
+        local = self._match(TokenType.LOCAL) is not None
+        global_ = self._match(TokenType.GLOBAL) is not None
+        session = self._match(TokenType.SESSION) is not None
+
```

Comment:
expression = self._parse_equality() , let's handle the TO later, but this is not the way, i'll have to fiddle with it